push-messaging-and-notifications/main.js

```diff
@@ -1,23 +1,42 @@
 'use strict';
 
-var API_KEY = '<YOUR API KEY>';
+var API_KEY = window.GoogleSamples.Config.gcmAPIKey;
+var GCM_ENDPOINT = 'https://android.googleapis.com/gcm/send';
 
 var curlCommandDiv = document.querySelector('.js-curl-command');
 var isPushEnabled = false;
 
 function sendSubscriptionToServer(subscription) {
-  // TODO: Send the subscription.subscriptionId and 
-  // subscription.endpoint to your server and save 
-  // it to send a push message at a later date
+  // TODO: Send the subscription.endpoint
+  // to your server and save it to send a
+  // push message at a later date
   console.log('TODO: Implement sendSubscriptionToServer()');
 }
 
-function showCurlCommand(subscription) {
+// NOTE: This code is only suitable for GCM endpoints,
+// When another browser has a working version, alter
+// this to send a PUSH request directly to the endpoint
+function showCurlCommand(pushSubscription) {
   // The curl command to trigger a push message straight from GCM
-  var subscriptionId = subscription.subscriptionId;
-  var endpoint = subscription.endpoint;
+  if (pushSubscription.endpoint.indexOf(GCM_ENDPOINT) !== 0) {
+    window.Demo.debug.log('This browser isn\'t currently ' +
+      'supported for this demo');
+    return;
+  }
+
+  var mergedEndpoint = pushSubscription.endpoint;
```

Comment:
See my previous comment—I think it would be a good idea to pre-concatenate the two values if needed and then pass in the full string to this function.